### PR TITLE
Syntax pass

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- Replace `Object.assign` with spread operators
- Use optional chaining

### CODE REVIEW

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js [hunk 1]
```diff
@@ -125,15 +126,9 @@ export default class ScreenGridLayer extends GridAggregationLayer {
...
-    if (aggregationBuffer) {
-      aggregationBuffer.delete();
-    }
-    if (maxBuffer) {
-      maxBuffer.delete();
-    }
-    if (maxTexture) {
-      maxTexture.delete();
-    }
+    aggregationBuffer?.delete();
```
Comment (ibgreen) [inline modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js:129]: Huh, I have only seen this used in function calls like `aggregationBuffer?.delete?.();`

I don't quite get how it works...

Does it equal to `(aggregationBuffer?.delete)();` (which could crash) or `aggregationBuffer?.(delete)());`

**Review by ibgreen — APPROVED** [review]
Are the `?.` supported in node 12? I recall having some issues.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js (hunk 1 above)
Comment (Pessimistress) [reply modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js:129]: `aggregationBuffer?.delete?.()` checks if `delete` exists before calling it.

In this particular context, if `aggregationBuffer` exists, it's always a Buffer object.

**Comment by Pessimistress** [comment]
Optional chaining is supported in Node>=14.

File: modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js (hunk 1 above)
Comment (ibgreen) [reply modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer.js:129]: I was worrying about the case where it doesn't exist - it wasn't clear to me that the `.?` construct in the middle would prevent evaluation of the function call operator at the end. I assumed it only applied to the field lookup. But I assume it does, that is great.

**Pull request merged**